_projects/enhance-and-improve-coala-community-and-projects-website.md

```diff
@@ -0,0 +1,160 @@
+---
+collaborating_projects:
+  - coala
+desc: "Enhance and Improve the architecture of coala Community and Projects
+website with new API Endpoints in webservices."
+difficulty: hard
+initiatives:
+  - GSoC
+issues:
+  - "https://github.com/coala/projects/issues/691"
+markdown: enhance-and-improve-coala-community-and-projects-website.md
+mentors:
+  - damngamerz
+  - sks444
+name: "Enhance websites architecture"
+requirements:
+  - "At least one patch accepted to coala community, projects and webservices
+  repository."
+  - "The applicant should be familiar with AngularJS, Django and Python."
+tags:
+  - community
+  - projects
+  - webservices
+  - AngularJS
+  - Django
+  - Python
+---
+### Overview
+The motivation here, is to enhance and improve the websites ([community](https://community.coala.io) and
+[projects](https://projects.coala.io)) for the coala community, for newcomers
+who wish to start contributing to coala open-source organization and for other
+organizations. The hurdle that newcomers mainly come across is the invitation
+to the organization before making any contributions, so this project adds a
+form for newcomers who wish to join the coala community. Also, to give a
+motivation to existing contributors, coala has created a `Promotion form` which
+will promote them to `coala Developer` status. To reduce the workload for
+maintainers, this project includes the development of form with some
+rudimentary checks that will promote the coala-newcomer to coala-developer.
+Moreover, this project adds major enhancements to websites that will help
+newcomers as well as developers to know more about this community and about
+what they have contributed to it.
+
+### Goals
+
+* Add calendar app to show timeline for various events
+* Add a one-stop tab for all google forms
+* Add contributors geolocation
+* Add *Login/SignUp* using GitHub
+* Add organization teams
+* Advanced filters
+* Automate welcoming process for newcomer
+* Automate newcomer promotion
+* Create new database models to manage data
+* Ease navigatibility in websites
+* Improve the **UI/UX of websites** to make more interactive
+* Motivating newcomers & developers with -
+  * modifications in Gamification Leaderboard
+  * adding a tab of **GSoC** in projects website
+* New coala-community logo
+* Contributors profile page
+* Admin/Maintainers dashboard
+
+### Milestones
+
+#### PREPARATION/BONDING
+
+* Detailed research on what **new cron jobs** or **API endpoints** need to be
```

Comment:
Before adding `new cron jobs` we need to fix the old ones, landing website is down right now, figure out how to fix the same. We might need to have CD in the landing repo.